src/lib/types/option.ts

```diff
@@ -57,7 +57,10 @@ export interface IAuthOption {
     enableApiToken: boolean;
     type: IAuthType;
     customAuthHandler?: Function;
-    createAdminUser: boolean;
+    initialAdminUser: {
```

Comment:
This new option makes sense to me. If you set both this and `createAdminUser` at the same time the values in this option should dictate the result. 

The only edge case I can think of is if you set `createAdminUser=false` and defines values for `initialAdminUser`. I this case my suggestion would be that we will not try to create the admin user.

What do you think?

Reply:
As stated in the other comment I'm guessing `createAdminUser === false` should take precedence, but ideally one shouldn't define both. Perhaps adding a runtime check to warn/throw if `createAdminUser: false` and `initialAdminUser: 'non-null'` are both explicitly passed? Also, I think if the env variables for username and passwords are present, this should force `createAdminUser` to be true. If people pass both a custom object AND use the env variables, I think we'd be wise to consider throwing an error, unless we can reliably resolve these and make the precedence clear in the docs.